report_data.py

```diff
@@ -235,19 +308,21 @@ def process_sales(self, df_sales):
             # to compute the "Kapitalertrag"
             self.add_fees(row, f"Selling {row.symbol}")
 
-            currency, new_forex = FIFOForex.from_share_sale(row)
+            _, new_forex = FIFOForex.from_share_sale(row)
             self.held_forex[row.currency].push(new_forex)
 
     def process_forex_to_eur(self, df_forex_to_eur):
         # When doing a currency conversion, you convert the USD you possess into the equivalent amount of EUR.
         # This doesn't include the fee you pay for the transfer, that just vanishes in the original denomination
-        for row_idx, row in df_forex_to_eur.iterrows():
+        for _, row in df_forex_to_eur.iterrows():
             sold_currency = row.currency
-            self.held_forex[sold_currency].pop(row.fees, row.date)  # remove fees
-            tmp = self.held_forex[sold_currency].pop(row.net_amount, row.date)
+            fees = to_decimal(row.fees)
```

Comment:
Many places there's the `to_decimal` function and it seems a bit fragile to find all places to replace it properly. If it's not happened, there's no error message, it "just" gets back to float.

What do you think about converting everything into Decimal during XLSX reading? So the DataFrame would already contain Decimals instead of floats.

Thorough the program run it'd be Decimals, then, at the very end, at the export would get back to floats again.